segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexRequestProcessor.java

```diff
@@ -94,7 +112,9 @@ private static Map.Entry<Long, Long> applySearch(StreamSegmentStore store, Strin
                     return entry.getOffset();
                 case Future:
                 case EndOfStreamSegment:
+                    throw new IllegalStateException(String.format("Unexpected size of index segment of type: %s was encountered for segment %s.", firstElement.getType(), segment));
                 case Truncated:
+                    throw new SegmentTruncatedException(String.format("Segment %s no longer exists.", segment));
                 default:
```

Comment:
This is an enum. I believe if the switch is exhaustive we can delete the default case.

Reply:
Removed the default case and also SearchFailedException.

Reply:
Done.